src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -121,15 +121,56 @@ bool DefaultOTARequestorDriver::ProviderLocationsEqual(const ProviderLocationTyp
     }
 }
 
-void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error) {}
+void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error)
+{
+    if (error == CHIP_NO_ERROR)
+    {
+        return;
+    }
+
+    switch (state)
+    {
+    case UpdateFailureState::kUnknown:
```

Comment:
This failure state can occur in `DefaultOTARequestor::NotifyUpdateApplied`. It doesn't make sense to start the query all over again if there's a failure there.

Reply:
We're retrying with the next provider in the list, if any.  Not retrying with the same provider here.  Do you think we should not retry with the next provider in this case?

Reply:
At this point, we had just finished successfully updating the software image (which is why we're sending the notify). Why do we want to start retrying on another provider since we literally just upgraded?

Reply:
Okay, so if we don't care that NotifyUpdateApplied() failed when updating software image, then we can just ignore this error and set Periodic Query timer.